src/storage/src/hummock/local_version_manager.rs

```diff
@@ -213,16 +215,33 @@ impl LocalVersionManager {
     /// Updates cached version if the new version is of greater id.
     /// You shouldn't unpin even the method returns false, as it is possible `hummock_version` is
     /// being referenced by some readers.
-    pub fn try_update_pinned_version(&self, newly_pinned_version: HummockVersion) -> bool {
+    pub fn try_update_pinned_version(
+        &self,
+        last_pinned: Option<u64>,
+        pin_resp: (bool, Vec<HummockVersionDelta>, HummockVersion),
+    ) -> bool {
+        let old_version = self.local_version.upgradable_read();
+        if let Some(last_pinned_id) = last_pinned {
+            if old_version.pinned_version().id() != last_pinned_id {
+                return false;
+            }
+        }
+        let newly_pinned_version = if pin_resp.0 {
+            let mut version_to_apply = old_version.pinned_version().version();
```

Comment:
Seems this line has introduced a constant overhead of version copy?
We can defer this cost until seeing old_version.pinned_version().id() < new_version_id.

Reply:
> Seems this line has introduced a constant overhead of version copy, even when old_version.pinned_version().id() >= new_version_id.

fixed